lib/AST/DocComment.cpp

```diff
@@ -426,24 +426,49 @@ class CommentProviderFinder final {
     return std::nullopt;
   }
 
+  /// Check if there is an inherited protocol that has a default implementation
+  /// of `VD` with a doc comment.
   std::optional<ResultWithDecl> findDefaultProvidedDecl(const ValueDecl *VD) {
```

Comment:
This is just duplicating logic in the witness checker. Default witnesses are recorded in the ProtocolDecl.

See TypeChecker::inferDefaultWitnesses() and ProtocolDecl::getDefaultWitness().

Reply:
That’s not sufficient though. We aren’t dealing with protocol requirements here, just default methods on a protocol.

For example in the following test, `InheritedProtocol.hello()` needs to inherit the documentation from `BaseProtocol.hello` but there is no protocol requirement that `hello()` can be a witness for. Any suggestion of how to implement that without my implementation?

```swift
protocol BaseProtocol {}
extension BaseProtocol {
  /// Say hello
  func hello() { }
}

protocol InheritedProtocol: BaseProtocol {}
extension InheritedProtocol {
  func hello() { }
}
```